path/__init__.py

```diff
@@ -155,60 +157,6 @@ def __get__(self, instance, owner):
         )
 
 
-class matchers(object):
```

Comment:
I've resisted making this change, despite the TODO, because converting the module to a package may have implications for compatibility (such as packages that might vendor path.py and rely on it being a single module without any absolute imports). It's probably best to separate this change as at least a distinct commit.